### PR TITLE
Refactor HTTP Files and add Image Generation

### LINKED ISSUE #47: Suggestion: Move HTTP files into Subfolder and Refactor
I'd like to move all of the `.http` files into a subfolder, and then create more such files.

At the moment there are two files, and they live in the root folder for the repo. The `test-aoai.http` file is current 286 lines long, and will grow as we want to test/call more of the Azure OpenAI API.

The reason for this is that they consume the `.env` file which is also in the root, and the REST Client extension only supports using a `.env` from the same folder.

My suggestion is...

1. Create folder `./http`
2. Move `.http` files into this new folder
3. Create a symlink using `ln -s ../.env ./env` within that folder, to make the `.env` file in the root of the repo available to the `.http` files
4. Finally, split out the 2 current `.http` files into a set of smaller files (one per API or area of OpenAI API functionality)

Suggested splitting `test-aoai.http` into the following files:

- `completions.http`
- `chat-completions.http`
- `embeddings.http`
- `image-gen.http`

The only tricky bit is that the .env file does not exist in the repo until the user create is. That doesn't prevent us committing the symlink file into git, but it would be a dangling reference. 

Perhaps `make env-sym-link` or something similar could be added to the `Makefile`. OR, the docs could include a step to say "either create a symlink, or otherwise copy your .env file into the `http` folder".

I will assume this is a good idea, and will take the "document it" approach for now. I'll include this work into a PR. If we don't like the idea I can back it out.

### PR DESCRIPTION
**Pull request by martinpeck**
This PR refactors the .http files so that...

- one file per API (to reduce clutter/length of files)
- add "REST Client environments" so that its easy to point the HTTP files at either local or azure
- add new image gen calls to the suite of HTTP files, and the required env var to support this

Closes #47

### CODE REVIEW

**Comment by martinpeck** [comment]
There's some overlap in what I've changed here, and what @hannesne has updates/change in the docs related to "getting started". 

If this gets merged before #46 then #46 needs tweaking.
If #46 gets merged first (which might be easier) I can tweak the docs to reflect the new location/file names, plus make sure that the instructions including choosing the REST Client environment.

So...my suggestion is merge #46 first, and assume any approval of this PR includes fixing the docs up first.

**Comment by martinpeck** [comment]
This would close #47

File: docs/config.md [hunk 1]
```diff
@@ -53,6 +56,10 @@ The `.http` files for testing the endpoints also use the `.env` file to set the
 
 > Note: when running the simulator it will auto-generate an API Key. This needs to be passed to the API when making requests. To avoid the API Key changing each time the simulator is run, set the `SIMULATOR_API_KEY` environment variable to a fixed value.
 
+## Configuring Endpoints
+
+There are a numner of environemt variables that specify API endpoints. Each of these environement variables is named ending `_ENDPOINT`. For all such environment variables the format is `scheme://fqdn` ir `scheme://fqdn:port`. e.g. `http://localhost:5000` or `https://example.openai.azure.com`. You should **not** include a trailing forward slash in the value of the environment variable.
```
Comment (stuartleeks) [inline docs/config.md:61]:
```suggestion
There are a number of environment variables that specify API endpoints. Each of these environment variables is named ending `_ENDPOINT`. For all such environment variables the format is `scheme://fqdn` or `scheme://fqdn:port`. e.g. `http://localhost:5000` or `https://example.openai.azure.com`. You should **not** include a trailing forward slash in the value of the environment variable.
```

**Review by stuartleeks — APPROVED** [review]
LGTM - a couple of minor typos.

This is definitely cleaner than the previous approach - thanks

**Pull request merged**